### DISCUSSION TITLE
Project full of errors after uninstalling a NuGet package

### DISCUSSION THREAD

**Discussion by userandaname** [microsoft/WindowsAppSDK#3355] [Q&A]
Hi, I installed a package using NuGet (https://github.com/HavenDV/H.NotifyIcon) and I'm a complete noob and didn't know how to use it properly, so I uninstalled it. In doing so, all of MainWindow scripts (MainWindow.xaml and .xaml.cs) is FULL of errors and it can't find Microsoft.UI which might've been deleted cos of the uninstall and I've made a new test project and it had the same issue, reinstalled VS and still not fixed. The design tab also won't work saying "Some assembly references are missing." I have no idea how to fix this at all!

Is this fixable?

![1](https://user-images.githubusercontent.com/81559574/214117269-8d44b4f5-b67a-4ece-b4ff-42094e583986.png)
![2](https://user-images.githubusercontent.com/81559574/214117273-eb76485d-5347-4676-aa08-557b375b66ae.png)
![3](https://user-images.githubusercontent.com/81559574/214117276-35485bff-be28-432d-a253-c11cddde56bb.png)
![4](https://user-images.githubusercontent.com/81559574/214117283-c4c28e6c-4e2d-4f59-8771-1bc29cdd9cb5.png)

**Comment by DarranRowe** [comment]
Did you actually try rebuilding the project? A quirk of the Visual Studio build system is that code generation and other things like that happens only when you build, not when the intellisense compiler runs.
Also, what nuget packages do you currently have referenced in the project?

**Reply by userandaname** [reply]
The only nuget packages I have are the default WinUI3 packages, the ones that corrupted mine have been uninstalled, I'm not sure how to rebuild my project yet, how do you do that?

I also have a new error in MainWindow.xaml: "Package Microsoft.WindowsAppSDK, version 1.2.221209.1 was not found. It might have been deleted since NuGet restore. Otherwise, NuGet restore might have only partially completed, which might have been due to maximum path length restrictions."

![nuget](https://user-images.githubusercontent.com/81559574/215286363-b57c95b8-dac4-4507-8858-5aee845992bc.png)

**Reply by DarranRowe** [reply]
Rebuilding is a nice and easy F7 or Ctrl+Alt+F7. You should also be able to find it in the Build menu:

![Screenshot 2023-01-28 193646](https://user-images.githubusercontent.com/52577874/215287556-84aa5d72-afe2-4748-ad11-48d5fbedadc1.png)

As for any error messages, ignore them for now until you build the project. The build should cause things to be downloaded again if they are missing.

**Reply by userandaname** [reply]
That didn't seem to do very much as all the errors are still there and the rebuild took somewhere around 3 seconds so I'm not sure whether it worked or not
